addons/edi_sale_notifier/models/edi_sale_request_document.py

```diff
@@ -0,0 +1,30 @@
+"""EDI sale order request documents"""
+
+from odoo import models
+
+
+class EdiSaleRequestDocument(models.AbstractModel):
+    """Extend ``edi.sale.request.document`` to generate a notifier suffix"""
+
+    _inherit = "edi.sale.request.document"
+
+    def report_invalid_records(self, doc):
+        """Identify a partial success and generate a suffix for the notifier to use
+
+        This function is only called if fail_fast is disabled.
+        """
+        PartnerRecord = self.partner_record_model(doc)
+        SaleLineRequestRecord = self.sale_line_request_record_model(doc)
+        SaleRequestRecord = self.sale_request_record_model(doc)
+
+        res = super().report_invalid_records(doc)
+        error_domain = [('doc_id', '=', doc.id), ('error', '!=', False)]
+        lines = SaleLineRequestRecord.search(error_domain)
+        orders = SaleRequestRecord.search(error_domain)
+        partners = PartnerRecord.search(error_domain)
+        if any([lines, orders, partners]):
+            doc.notifier_subject_suffix = " - with issues"
+        else:
```

Comment:
Is this needed? Won't `notifier_subject_suffix` be blank by default?

Reply:
I put the else as we want to clear any messages that may have been set. Perhaps on an earlier unsuccessful execution